src/alloc_pool.c

```diff
@@ -511,7 +513,9 @@ void qd_dealloc(qd_alloc_type_desc_t *desc, qd_alloc_pool_t **tpool, char *p)
     sys_mutex_unlock(desc->lock);
 }
 
-
+// use-after-free; no way to check if memory is freed before
+// accessing it from `qd_alloc_deref_safe_ptr`
+ATTRIBUTE_NO_SANITIZE_ADDRESS
```

Comment:
This macro should be also made to work for TSan. TSan does check for some invalid accesses and it barfs on this one.